extensions/amp-story/1.0/amp-story-open-page-attachment.js

```diff
@@ -152,11 +156,43 @@ const renderOutlinkPageAttachmentUI = (
     openAttachmentEl.setAttribute('href', attachmentHref);
   }
 
+  // Getting elements
+  let {barLeft, barRight, chipEl, ctaLabelEl} = htmlRefs(openAttachmentEl);
+
   // Setting theme
-  openAttachmentEl.setAttribute('theme', attachmentEl.getAttribute('theme'));
+  const themeAttribute = attachmentEl.getAttribute('theme');
+  openAttachmentEl.setAttribute('theme', themeAttribute);
 
-  // Getting elements
-  const {chipEl, ctaLabelEl} = htmlRefs(openAttachmentEl);
+  const accentColor = attachmentEl.getAttribute('cta-accent-color');
+  if (themeAttribute === "custom") {
+    if (attachmentEl.getAttribute('cta-accent-element') === "background") {
+      setImportantStyles(ctaLabelEl, {
+        'color': contrastColor
+      });
+      setImportantStyles(chipEl, {
+        'background-color': accentColor
+      });
+      setImportantStyles(barLeft, {
+        'background-color': accentColor
+      });
+      setImportantStyles(barRight, {
+        'background-color': accentColor
+      });
+    } else {
+      setImportantStyles(ctaLabelEl, {
+        'color': accentColor
+      });
+      setImportantStyles(chipEl, {
+        'background-color': contrastColor
+      });
+      setImportantStyles(barLeft, {
+        'background-color': contrastColor
+      });
+      setImportantStyles(barRight, {
+        'background-color': contrastColor
+      });
```

Comment:
Please set one CSS variable and use it for the `background-color` properties directly in the CSS files, this will simplify the JavaScript